### PR TITLE
MM-47109 : Migrate components/list_modal.jsx to Typescript

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
Migrate components/list_modal.jsx to Typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21061
JIRA: https://mattermost.atlassian.net/browse/MM-47109

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@AbhinavVihan can you please CI for us to initiate the review

**Comment by AbhinavVihan** [comment]
@M-ZubairAhmed, Its ready for review. Thanks

File: components/list_modal.tsx [hunk 1]
```diff
@@ -1,87 +1,46 @@
...
-         */
-        numPerPage: PropTypes.number,
-
-        /**
-         * show (optional) a boolean setting to hide the modal via props rather then unmounting it.
-         */
-        show: PropTypes.bool,
-    };
+type Props = {
+    titleText: string;
```
Comment (m1lt0n) [inline components/list_modal.tsx:14]: Based on the comments, also `titleText`, `show` and `searchPlaceholderText` are optional. Below they are marked as required.

**Review by m1lt0n — CHANGES_REQUESTED** [review]
Looks good, left a couple of comments about optional props.

**Review by m1lt0n — APPROVED** [review]

File: components/list_modal.tsx [hunk 2]
```diff
@@ -1,87 +1,46 @@
...
 
 import LoadingScreen from 'components/loading_screen';
+import {Group} from '@mattermost/types/groups';
 
 export const DEFAULT_NUM_PER_PAGE = 50;
 
-export default class ListModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
```
Comment (sinansonmez) [inline components/list_modal.tsx:16]: As I was informed previously, these comments are picked up by some tool for documentation. For this reason, I would keep the comments

**Review by sinansonmez — CHANGES_REQUESTED** [review]
Thanks for the contribution. I just have one minor comment related to deleted comments

**Review by sinansonmez — APPROVED** [review]
LGTM thanks

File: components/list_modal.tsx (hunk 2 above)
Comment (sinansonmez) [reply components/list_modal.tsx:16]: Thanks for adding back the comments

**Comment by jgilliam17** [comment]
@m1lt0n can you please approve the last workflow and merge? Thanks 🙂

**Comment by jordanafung** [comment]
@AbhinavVihan - Thank you for participating in Hacktoberfest!
Please fill out [this form](https://docs.google.com/forms/d/1s0xarmIVqXcut1Z7waaKqrWbuwCpGg4s9lA-bChO89Y/) if you'd like to get your Hacktoberfest Stickers!

**Comment by jordanafung** [comment]
Hi @AbhinavVihan 
You were selected as one of our Top Contributors for Hacktoberfest!
Please fill out [this form](https://docs.google.com/forms/d/11pLB3pF1CmOslYqHr7c7bxFWU6QAEAJuiaKgHD1vJtg) to select your swag as thanks!
Thank you for all you do and for making Mattermost even better!

**Pull request merged**